### PR TITLE
ROX-25519: Avoid immediately removing orphaned node cves - Part 1

### PR DESCRIPTION
**Pull request by charmik-redhat**
### Description

This is the first part of a 3 part change to prevent immediate removal of orphaned Node CVEs
Part-1: Adds proto changes, generate store and datastore. Adds environment variables to enable/disable these changes.

Part-2: Will update node store implementation to move orphaned CVEs to `orphaned_node_cves`  table instead of removing them. Will also add logic to restore orphaned CVEs if they are detected again.

Part-3: Will add implementation to prune orphaned CVEs that have not been restored for a set period of time.

### User-facing documentation

- [X] CHANGELOG update is not needed
- [X] documentation is not needed

### Testing and quality

- [X] the change is production ready: the change is GA or otherwise the functionality is gated by a feature flag
- [ ] CI results are inspected

#### Automated testing

- [X] added unit tests

#### How I validated my change

- Manually verified that the generated schema is correct.
- Autogenerated unit tests for the storage layers and added unit tests for datastore layer.

### CODE REVIEW

File: pkg/env/vulnerability_management.go [hunk 1]
```diff
@@ -13,4 +13,11 @@ var (
 	// ImageCVEEdgeCustomJoin is to enable/disable the adding custom join with image_cve_edges table.
 	// It is a temporary workaround to fix image_cve_edge search issue (ROX-17252)
 	ImageCVEEdgeCustomJoin = RegisterBooleanSetting("ROX_IMAGE_CVE_EDGE_CUSTOM_JOIN", true)
+
+	// OrphanedCVEsKeepAlive is to enable/disable preserving orphaned CVEs for a short interval before removing them
+	// from storage
+	OrphanedCVEsKeepAlive = RegisterBooleanSetting("ROX_ORPHANED_CVES_KEEP_ALIVE", false)
```
Comment (dashrews78) [inline pkg/env/vulnerability_management.go:19]: Should we default to true?

Comment (charmik-redhat) [reply pkg/env/vulnerability_management.go:19]: I think we keep it disabled and only have the affected customer turn it on, at least for the patch.

**Pull request merged**